ReSharper.FSharp/src/FSharp/FSharp.Psi.Daemon/src/Stages/PatternTypeHintsStage.fs

```diff
@@ -21,28 +22,39 @@ open JetBrains.ReSharper.Plugins.FSharp.Psi.Services.Util.TypeAnnotationsUtil
 open JetBrains.TextControl.DocumentMarkup.Adornments.IntraTextAdornments
 
 type private NodesRequiringHints =
-    { TopLevelNodes: VisibilityConsumer<ITreeNode>; LocalNodes: VisibilityConsumer<ITreeNode> } with
+    { TopLevelNodes: VisibilityConsumer<ITreeNode>
+      LocalNodes: VisibilityConsumer<ITreeNode>
+      OtherPatterns: VisibilityConsumer<ITreeNode> } with
 
     member x.HasVisibleItems =
         x.TopLevelNodes.HasVisibleItems ||
-        x.LocalNodes.HasVisibleItems
+        x.LocalNodes.HasVisibleItems ||
+        x.OtherPatterns.HasVisibleItems
 
 type private FSharpTypeHintSettings =
-    { TopLevelMembers: PushToHintMode; LocalBindings: PushToHintMode } with
+    { TopLevelMembers: PushToHintMode
+      LocalBindings: PushToHintMode
+      OtherPatterns: PushToHintMode } with
```

Comment:
Is there a reason `with` keyword is required here? Should we have an analyzer/QF for removing it?